live-build/misc/ansible-roles/appliance-build.minimal-common/tasks/main.yml

```diff
@@ -369,13 +369,57 @@
     regexp: '^GRUB_CMDLINE_LINUX_DEFAULT='
     line: 'GRUB_CMDLINE_LINUX_DEFAULT="$GRUB_CMDLINE_LINUX_DEFAULT crashkernel=1024M-:512M"'
 
-# Panic on ZFS assertion failures
-- lineinfile:
-    path: /etc/modprobe.d/zfs.conf
-    create: yes
-    line: 'options spl spl_panic_halt=1'
+- copy:
+    dest: /etc/modprobe.d/10-zfs.conf
+    mode: 0644
+    content: |
+      #
+      # This file should not be directly modified by support, as these
+      # changes would be removed on upgrade. In order to make custom
+      # tunable changes for specific customer installations, please edit
+      # the file (creating it if necessary): /etc/modprobe.d/20-zfs.conf
+      #
+
+      #
+      # Panic on ZFS assertion failures; otherwise it's possible for these
+      # assertion failures to only halt the thread that failed the assert,
+      # which can then put the entire system into a "hung" state, since
+      # we're running a rootfs on ZFS.
+      #
+      options spl spl_panic_halt=1
+
+      #
+      # We increase the max memory limit for ZFS channel programs to 512MB
+      # to give us a wide buffer for development of channel programs. We
+      # were using upwards of 70MB during internal testing, which is too
+      # much with the default limit of 100MB.
+      #
+      options zfs zfs_lua_max_memlimit=536870912
+
+      #
+      # On Linux, SEEK_FILE and SEEK_HOLE are inaccurate if the file is
+      # dirty; in which case, they say that there are no holes. To get the
+      # same behavior as illumos, we need to configure this tunable.
+      #
+      options zfs zfs_dmu_offset_next_sync=1
+
+      #
+      # The following changes are made to match the values for these same
+      # tunables, that we use on our illumos-based DelphixOS.
+      #
```

Comment:
We'll probably want to improve this comment at some point, since it basically is just saying "this is the way we've always done it".  But this is fine for now.

Reply:
Yea. I just pulled this comment from the commit message on the zfsonlinux repository that made the change.

Since I'm already here, I'll do a little more legwork and look at the commit message and/or bug for our illumos repository, and see if I can add some more details here.